lang/java/avro/src/main/java/org/apache/avro/generic/GenericData.java

```diff
@@ -69,6 +69,16 @@
 
   private static final GenericData INSTANCE = new GenericData();
 
+  private static final Map<Class<?>, String> PRIMATIVE_DATUM_TYPES = new IdentityHashMap<>();
```

Comment:
Any reason to use IdentityHashMap in favour of HashMap ?

Reply:
It's faster to use `IdentityHashMap` as it uses the primitive `==` method instead of `equals`.  When coming up with this solution, I did some research and from everything I can find, it seems that as long as the classes comes from the same class loader (as is the case here since it's populated by the class itself and not an outside reference), then this should be perfectly acceptable.

Reply:
That's true, but `java.lang.Class` does not override `#equals()`, so it inherits `Object#equals()` which uses `==`.

Reply:
They're equivalent -- let's go with @belugabehr 's choice!